### ISSUE TITLE
cover description always recommended?

### ISSUE THREAD

**Issue by mattgarrish** [w3c/pub-manifest#145]
I noticed that we recommend both `title` (this is a mistake I'll fix, as it should be `name`) and `description` for all cover images.

Recommending `name` for alt text seems fine, but not every cover needs a description, only ones with complex images. Can we reword this more along the lines of:

> If the cover is in an image format, a `name` SHOULD be provided. User agents can use this property to provide alternative text for the image. The `description` property can also be used to provide an extended description, when necessary.

/cc @GeorgeKerscher @avneeshsingh @clapierre

**Comment by clapierre** [comment]
I think I like this but personally would rather the name be a "MUST" with the description being a "MAY".  What would be a use-case where a cover image would not have a name?
WCAG-A 1.1.1 requires all images be described, so why allow an author to not be WCAG-A compliant?
Unless you are saying there is a scenario where a cover image is purely decorative?

**Comment by GeorgeKerscher** [comment]
If the only useful information is the name, then no alt text/description would be provided, right?

 

Best

George

**Comment by mattgarrish** [comment]
> What would be a use-case where a cover image would not have a name?

I think the "should" approach is that we're not trying to enforce WCAG conformance at this level. We recommend alt text for all publications, but you can choose not to provide it depending on your circumstances. That was the comprise that led to a recommendation, as I recall. It definitely becomes a must when conformance has to be attained, I agree.

**Comment by mattgarrish** [comment]
> If the only useful information is the name, then no alt text/description would be provided, right?

The name property is used as the alt text as for raw images, so if you specify the name you don't need to do anything else. We're recommending that you always name your images so that information can be used, but it seems unnecessary to also recommend an extended description for all images when extended descriptions aren't often applicable.

**Comment by avneeshsingh** [comment]
Slightly modified version is as follows. Please feel free to improve it.
If the cover is in a format  of meaningful image, it is strongly recommended to provide a name, to enable User agents to provide alternative text for the image. 
The description property can also be used to provide an extended description for a complex image for which alttext may not be sufficient.

**Comment by mattgarrish** [comment]
>  meaningful image

This part is potentially problematic for verification, as only a human can determine meaning. What if we move the problem to WCAG, and better explain that we want all covers described, along these lines:

> If the cover is an image (whether embedded in an HTML resource or not), it is strongly advised to follow Success Criterion 1.1.1 of [WCAG] for the provision of alternative text and extended descriptions. For image formats that don't provide the ability to embed this information, the `name` and `description` properties of a `LinkedResource` can be used to provide alternative text and extended descriptions, respectively.

We won't be able to have a generic warning from user agents, but I think this kind of checking ultimately falls to more specialized tools, like Ace. There's still the possibility, too, that a user agent could send an info message when name isn't set.

**Comment by mattgarrish** [comment]
I wonder if we could also make a recommendation, like:

> If alternative text is needed and not provided, user agents SHOULD use the word 'cover' followed by the `name` of the publication.

Or something along those lines.

**Comment by clapierre** [comment]
I like this idea Matt.

**Comment by laudrain** [comment]
I like it also. 
BTW that’s exactly what we’re asking to our suppliers today for the cover image of all EPUB3 files: the alt attribute should contained the localized word for ‘cover’ (‘couverture’ in French) followed by the title and the author of the publication. 
In fact, it is the text on the cover.

**Comment by deborahgu** [comment]
I don't like saying "complex" as a criterion for alt text, I like "meaningful" or "usefully identifiable".

* [Complex but meaningless cover](https://images-na.ssl-images-amazon.com/images/I/61q21-jvw%2BL._SX342_QL70_.jpg)
* [Very simple, very meaningful cover](https://images-na.ssl-images-amazon.com/images/I/31VTPC7KRSL._SY291_BO1,204,203,200_QL40_.jpg)
* [Meaningless cover but sendmail administrators call this "the fruitbat book"](https://covers.oreillystatic.com/images/9780596510299/lrg.jpg)
* [Meaningless cover but computer science students call this "the Dragon book"](http://www.cs.nott.ac.uk/~psztxa/g52mal/book_pics/dragon1st.jpg)

it sounds like we are going to link to external documentation and make a WCAG recommendation for a cover page technique, and if we do so, we should be sure to recommend useful or meaningful iinstead of "complex".

**Comment by dauwhe** [comment]
Is having metadata that is just the word "cover" plus the book title (and author?) actually helpful? It's not adding information to the system.

**Comment by clapierre** [comment]
It is, as compared with just hearing "Image", especially if that is the only item that is exposed when displaying a list of titles to the end user.

**Comment by dauwhe** [comment]
> It is, as compared with just hearing "Image", especially if that is the only item that is exposed when displaying a list of titles to the end user.

Are there developers who are so evil that they will do this, yet so thoughtful that they still include the alt text? :)

**Comment by iherman** [comment]
This issue was discussed in [a meeting](https://www.w3.org/publishing/groups/publ-wg/Meetings/Minutes/2019/2019-11-04-pwg#section1-4).

- No actions or resolutions
<details><summary><i class="differentiate">View the transcript</i></summary>
cover description always recommended?<br/>
Matt Garrish: <a href="https://github.com/w3c/pub-manifest/issues/145">https://github.com/w3c/pub-manifest/issues/145</a><br/>
Matt Garrish: yes, 145… i think this is coming to a solution… that we recommend both a name and a description for an image…<br/>
… like saying we need alt text and description for all images, which is not always true…<br/>
… recommend we point to WCAG… would be ideal if we could agree on this now, or wrap it up in the next couple of days<br/>
… not all covers are meaningful and need to be described or need extended descriptions…<br/>
… this becomes more of a guidance to the user and the author<br/>
Garth Conboy: <a href="https://github.com/w3c/pub-manifest/issues/145#issuecomment-549448461">https://github.com/w3c/pub-manifest/issues/145#issuecomment-549448461</a><br/>
Deborah Kaplan: flashing back to the long conversation we had about this when it first came up… i agree with the wording that you said should be changed… but a particular trick is that the problem is not any image needing to be described, but particular complex images<br/>
… we need to make a recommendation for how the UA should handle this information… and it should follow the standard rules…<br/>
… whether by linking to existing docs in an existing spec or stating here that meaningful images should be described… we need to have it somehow<br/>
Matt Garrish: I fully agree, that was my intention that we’d link… but i’m wondering about how much we should repeat… I hate doing that because it tends to introduce ambiguity… but we could explicitly point what we mean about “meaningful images” to WCAG<br/>
Deborah Kaplan: I would be ok with the approach, but this is a difficult problem because this is a domain-specific aspect of the art of alt text that people don’t understand… people don’t usually understand that the cover contains metadata and images that are usually decorative, but not always<br/>
Bill Kasdorf: this isn’t just about books, is it?<br/>
Deborah Kaplan: WCAG does not explain this carefully… it’s a set of guidelines, not about the art… it’s domain-specific enough that it could use a little help<br/>
… this is why I don’t like the concept of “complex” (fractals are complex, but decorative, for example)<br/>
Matt Garrish: I completely agree with dkaplan, but we should raise it with WCAG about adding a technique, bringing publishing issues across W3C<br/>
Deborah Kaplan: matt: +100000090000 for technique for book covers!!!<br/>
Charles LaPierre: +1 Agreed on book cover techniques!<br/>
Deborah Kaplan: also agreed with George: we called my CS book the dragon book, or sendmail the fruitbat book<br/>
George Kerscher: I have seen times where students refer to the book by the cover image… and we need enough information so that we can identify the book… and I agree with pointing to WCAG guidelines and bringing this up with them<br/>
Avneesh Singh: don’t know if WCAG would be ready to absorb it right away, cover is so specific to publishing… but this could go into best practices too… not to decide yet<br/>
Garth Conboy: this sounds like consensus…. mattg?<br/>
Matt Garrish: yes, this sounds like consensus, but we need to bring up a technique for covers with WCAG. These are the directions we can move forward with<br/>
Garth Conboy: great, sounds like at least rough consensus<br/>
</details>

**Comment by mattgarrish** [comment]
> Are there developers who are so evil that they will do this, yet so thoughtful that they still include the alt text? :)

Authors do it all the time... ;)

I think it's just a matter of consistency in providing guidance on how an alt text could be compiled. It's not strictly necessary we get to this level of detail, except perhaps to remind developers to think about accessibility when conceiving of uses for the cover.

The only required piece of metadata is the title, so that's why I suggested it. In the case of an auto-generated alt text, I don't think the rules or result should be terribly complicated.

**Comment by deborahgu** [comment]
> Are there developers who are so evil that they will do this, yet so thoughtful that they still include the alt text? :)

Goodness, yes. Sometimes it's that people *really* don't understand alt, but usually it's a CMS thing. I've seen CMS's that auto-populate alt with the filename 🤦🏽, or with the word "Image".

**Comment by mattgarrish** [comment]
If the revised text needs any additional improvements, please open a new issue.
